### PR TITLE
Group Details are Viewable

### LINKED ISSUE #21: Create UI and a view model that displays all the data for a Group
# High level description
- Make a View that displays all the information such as tasks, progress, and members for a specific group.
- Should look similar to screenshot below:
image

Note: You don't need to implement other screens like the submit task screen that opens up when you try to submit a task. You can also assume that someone else will implement functionality to navigate to this screen.

# Implementation details
- Create a view called GroupCardView (within Groups/Views) and a view model called GroupCardViewModel (within Groups/ViewModels) and fill them in with required information

### PR DESCRIPTION
**Pull request by sacchit25**
Group Details are Viewable

Created GroupView to display the content of the group.

Fixes #21 

https://github.com/user-attachments/assets/ca0494ae-f89e-4373-ab41-6a4261526134

### CODE REVIEW

File: Vineyard/Vineyard/GroupView.swift [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+//  GroupView.swift
+//  Vineyard
+//
+//  Created by Sacchit Mittal on 10/1/24.
+//
+
+import SwiftUI
+
+struct GroupView: View {
+    @Binding var group: Group
```
Comment (rahulon12) [inline Vineyard/Vineyard/GroupView.swift]: nit: this doesn't need to be a `@Binding`, at least right now
```suggestion
    let group: Group
```

File: Vineyard/Vineyard/GroupView.swift [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                    
+                    ForEach(group.people) {people in
+                        VStack(alignment: .leading, spacing: 10) {
+                            HStack {
+                                Image(systemName: "person.crop.circle")
+                                    .resizable()
+                                    .frame(width: 30, height: 30)
+                                Text("\(people.name)")
+                                Spacer()
+                                SwiftUI.ProgressView(value: 0.20)
```
Comment (rahulon12) [inline Vineyard/Vineyard/GroupView.swift]:
```suggestion
                                ProgressView(value: 0.20)
```

Comment (rahulon12) [reply Vineyard/Vineyard/GroupView.swift]: one more above as well

**Review by sankaet8 — CHANGES_REQUESTED** [review]
Rahul made some great points. The rest of the code looks good to me!

**Review by sankaet8 — APPROVED** [review]

**Pull request merged**